### DISCUSSION TITLE
A few thoughts about the new minimap

### DISCUSSION THREAD

**Discussion by vorlif** [wagtail/wagtail#9546] [General]
Hi devs,

thank you very much for your work and the many new improvements. I have currently looked at the new MiniMap and think it is excellent.  However, I would like to share a few thoughts with you on a few points. 

* In principle, I would like the MiniMap to be animated so that there is a smoother transition from display to disappearance. The sudden appearance and disappearance makes me a little uneasy. However, I am not a designer and generally Wagtail is sparing with animations. 

* If you move the mouse over the scrollbar, the MiniMap disappears again. This is quite fast, because the area for opening the MiniMap is relatively small. (It is also a "mini" map).

[Peek 2022-10-30 14-14.webm](https://user-images.githubusercontent.com/4333770/198880437-656ee48e-61d6-4698-a165-9248cc5808c9.webm)

* If you want to disable the comment function, you (I) quickly get into the MiniMap area and it overlays the disable button. Maybe it is also because I use a touchpad. 

[Peek 2022-10-30 14-18.webm](https://user-images.githubusercontent.com/4333770/198880568-24b292d2-06f9-4c80-93ba-4bc79795b15f.webm)

* In Chrome, sometimes an extra scrollbar appears.

![image](https://user-images.githubusercontent.com/4333770/198880751-f968b92b-98d5-4d1e-a4f4-c31c1e035c1a.png)

* Sometimes an "Open" arrow appears and sometimes it does not. (Chrome and Firefox)

![image](https://user-images.githubusercontent.com/4333770/198880940-2d70d0b2-62ce-4ef5-90b0-923133f8bf0e.png)

* For streamfields, a snippet of the content is displayed. I find this extremely useful for getting around better. Unfortunately, the content is rarely updated (when saving?) and is thus no longer up to date during editing. So I was sometimes confused where I am and where I have to go. 

![image](https://user-images.githubusercontent.com/4333770/198881452-c2267bb5-eae6-456a-89b2-a9c7da4fa3d7.png)

* I would like it if there was a way to display the MiniMap permanently.
* It would be nice if a click in the MiniMap would expand a collapsed field.  

These are all not problems or something that needs to be changed and the advantages of the MiniMap clearly outweigh them. I especially love the display when saving where something still needs to be filled in. But maybe someone finds that here still adjustments would be useful.

**Comment by thibaudcolas** [comment]
Hi @vorlif 👋 thank you for this feedback, we did take it into account in our most recent iterations on the minimap (yet to be released), just forgot to reply here.

We’re hoping to implement almost all of your suggestions, some in the next release (see #10055). The main thing which I’m uncertain about is:

> It would be nice if a click in the MiniMap would expand a collapsed field.

We did consider this, and it wasn’t very clear to us whether this would be the expected behavior for most users.
